invisible_cities/cities/detsim_test.py

```diff
@@ -0,0 +1,277 @@
+    result = detsim(**conf)
+
+    tables = ["pmtrd", "sipmrd",
+              "Run/eventMap", "Run/events", "Run/runInfo",
+              "MC/configuration", "MC/event_mapping", #"MC/hits", note that MC/hits are different
+              "MC/particles", "MC/sns_positions", "MC/sns_response",
+              "Filters/active_hits"]
+
+    with tb.open_file(PATH_OUT) as output_file:
+        with tb.open_file(modified_file_out) as modified_output_file:
+            for table in tables:
+                assert hasattr(output_file.root, table)
+                got      = getattr( modified_output_file.root, table)
+                expected = getattr(          output_file.root, table)
+                assert_tables_equality(got, expected)
+
+
+def test_detsim_buffer_times(ICDATADIR, output_tmpdir):
+
+    PATH_IN   = os.path.join(ICDATADIR    , "nexus_new_kr83m_fast.newformat.sim.h5")
+    PATH_OUT  = os.path.join(output_tmpdir, "detsim_test.h5")
+
+    physics_params = dict()
+    physics_params["ws"]                     = 39.2 * units.eV
+    physics_params["wi"]                     = 22.4 * units.eV
+    physics_params["fano_factor"]            = 0.15
+    physics_params["conde_policarpo_factor"] = 1.00
+    physics_params["drift_velocity"]         = 1.00 * units.mm / units.mus
+    physics_params["lifetime"]               = 12   * units.ms
+    physics_params["transverse_diffusion"]   = 1.20 * units.mm / units.cm**0.5
+    physics_params["longitudinal_diffusion"] = 0    * units.mm / units.cm**0.5
+    physics_params["el_gain"]                = 365
+    physics_params["el_drift_velocity"]      = 1000000 * units.mm / units.mus
+
+    buffer_params = dict()
+    buffer_params["length"]      = 800 * units.mus
+    buffer_params["pmt_width"]   = 100 * units.ns
+    buffer_params["sipm_width"]  =   1 * units.mus
+    buffer_params["max_time"]    =  10 * units.ms
+    buffer_params["pre_trigger"] = 100 * units.mus
+    buffer_params["trigger_thr"] = 0
+
+    # run over test file
+    conf = configure('detsim $ICTDIR/invisible_cities/config/detsim.conf'.split())
+    conf.update(dict(files_in      = PATH_IN,
+                     file_out      = PATH_OUT,
+                     run_number    = 0,
+                     event_range   = (1, 2),
+                     physics_params= physics_params,
+                     buffer_params = buffer_params))
+    result = detsim(**conf)
+
+    # compute max signal time
+    hits = load_dst(PATH_IN, "MC", "hits")
+    hits = hits[hits["event_id"]==1]
+    z, time  = hits.z.values, hits.time.values
+    s2_times = z/physics_params["drift_velocity"] + time
+
+    s2_tmax = max(s2_times) + buffer_params["pre_trigger"]
+
+    with tb.open_file(PATH_OUT) as h5out:
+        pmtwf  = h5out.root.pmtrd.read()[0].sum(axis=0)
+
+    signal_bins = np.argwhere(pmtwf>0).flatten()
+
+    # start s1 time
+    tstart = signal_bins[0]*buffer_params["pmt_width"]
+    assert tstart == buffer_params["pre_trigger"]
+
```

Comment:
Maybe now you can simply assume that s2 will be after `pre_trigger + 2*mus` so set to s2s indices whatever is after that time. Then you can check that both start and end make sense. And do the same for sipms waveforms (without the s1 of course)